app/src/main/java/org/wikipedia/auth/AccountUtil.kt

```diff
@@ -125,7 +126,7 @@ object AccountUtil {
     }
 
     fun isUserNameTemporary(userName: String): Boolean {
-        return userName.startsWith("~")
+        return userName.length > 6 && userName[0] == '~' && userName[5] == '-' && userName.substring(1, 5).isDigitsOnly()
```

Comment:
Do you have the references for the username policy or naming rules?

Reply:
Sure: https://www.mediawiki.org/wiki/Help:Temporary_accounts#How_do_temporary_accounts_work?

Reply:
Thanks for the reference!